calibrators/marker_radar_lidar_calibrator/src/transformation_estimator.cpp

```diff
@@ -0,0 +1,188 @@
+    double calibrated_radar_to_radar_optimization_z =
+      calibrated_radar_to_radar_optimization_transformation.translation().z();
+    double calibrated_radar_to_radar_optimization_roll =
+      calibrated_radar_to_radar_optimization_rpy.x;
+    double calibrated_radar_to_radar_optimization_pitch =
+      calibrated_radar_to_radar_optimization_rpy.y;
+
+    if (
+      calibrated_radar_to_radar_optimization_z != 0.0 ||
+      calibrated_radar_to_radar_optimization_roll != 0.0 ||
+      calibrated_radar_to_radar_optimization_pitch != 0.0) {
+      RCLCPP_ERROR(
+        rclcpp::get_logger("marker_radar_lidar_calibrator"),
+        "The estimated 2D translation was not really 2D. Continue at your own risk. z=%.3f "
+        "roll=%.3f "
+        "pitch=%.3f",
+        calibrated_radar_to_radar_optimization_z, calibrated_radar_to_radar_optimization_roll,
+        calibrated_radar_to_radar_optimization_pitch);
+    }
+
+    calibrated_radar_to_radar_optimization_transformation.translation().z() =
+      (initial_radar_to_lidar_eigen_ * radar_optimization_to_lidar_eigen_.inverse())
+        .translation()
+        .z();
+  }
+
+  calibrated_radar_to_lidar_transformation_ =
+    calibrated_radar_to_radar_optimization_transformation * radar_optimization_to_lidar_eigen_;
+}
+
+void TransformationEstimator::estimateZeroRollTransformation()
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("marker_radar_lidar_calibrator"),
+    "Estimate 3D transformation with roll is always zero");
+
+  ceres::Problem problem;
+
+  Eigen::Vector3d translation = initial_radar_optimization_to_radar_eigen_.translation();
+  Eigen::Matrix3d rotation = initial_radar_optimization_to_radar_eigen_.rotation();
+  Eigen::Vector3d euler_angle = rotation.eulerAngles(0, 1, 2);
+
+  // params: x, y, z, pitch, yaw
+  std::vector<double> params = {
+    translation[0], translation[1], translation[2], euler_angle[1], euler_angle[2]};
+
+  std::string initial_params_str = std::accumulate(
+    std::next(params.begin()), params.end(),
+    std::to_string(params.front()),  // Initialize with the first element
+    [](const std::string & a, const auto & b) { return a + " " + std::to_string(b); });
+  std::string initial_params_msg = "initial params (x,y,z,pith,yaw): " + initial_params_str;
+  RCLCPP_INFO(
+    rclcpp::get_logger("marker_radar_lidar_calibrator"), "%s", initial_params_msg.c_str());
+
+  for (size_t i = 0; i < lidar_points_ocs_->points.size(); i++) {
+    auto lidar_point = lidar_points_ocs_->points[i];
+    auto radar_point = radar_points_rcs_->points[i];
+
+    Eigen::Vector4d radar_point_eigen(radar_point.x, radar_point.y, radar_point.z, 1);
+    Eigen::Vector4d lidar_point_eigen(lidar_point.x, lidar_point.y, lidar_point.z, 1);
+
+    ceres::CostFunction * cost_function = new ceres::AutoDiffCostFunction<SensorResidual, 3, 5>(
+      new SensorResidual(radar_point_eigen, lidar_point_eigen));
+    problem.AddResidualBlock(cost_function, NULL, params.data());
```

Comment:
nullptr

Reply:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/ef7043564d408baaaeb37eb085908c4e24a58333